stripe.go

```diff
@@ -118,17 +132,38 @@ func newAPIResponse(res *http.Response, resBody []byte) *APIResponse {
 	}
 }
 
+func newStreamingAPIResponse(res *http.Response, body io.ReadCloser) *StreamingAPIResponse {
+	return &StreamingAPIResponse{
+		Header:         res.Header,
+		IdempotencyKey: res.Header.Get("Idempotency-Key"),
+		Body:           body,
+		RequestID:      res.Header.Get("Request-Id"),
+		Status:         res.Status,
+		StatusCode:     res.StatusCode,
+	}
+}
+
 // APIResource is a type assigned to structs that may come from Stripe API
 // endpoints and contains facilities common to all of them.
 type APIResource struct {
 	LastResponse *APIResponse `json:"-"`
 }
 
+// APIStream is a type assigned to streaming responses that may come from Stripe API
+type APIStream struct {
+	LastResponse *StreamingAPIResponse `json:"-"`
```

Comment:
What does the JSON annotation here do?

Reply:
It will suppress the field from `json.Marshal`. This is appropriate for `APIResource` because APIResource is embedded in other structs and the user (or e.g. stripe-go's test suite) would expect `json.Marshal` to produce the same JSON that was returned from the stripe API and not to contain this weird extra LastResponse field, but I think it is not appropriate here as there should never have been any JSON from the stripe API to begin with.
```suggestion
	LastResponse *StreamingAPIResponse
```